src/main/java/org/cdlib/ill/report/excel/ReportWorkbookBuilder.java

```diff
@@ -25,156 +25,160 @@
 
 public class ReportWorkbookBuilder<T> {
 
-    private final XSSFWorkbook workbook;
-    private XSSFSheet dataSheet;
-    private XSSFSheet pivotTableSheet;
-    private XSSFPivotTable pivotTable;
-    private AreaReference source;
-    private final List<Field<T>> fields = new ArrayList<>();
-    private Collection<T> data;
-
-    private ReportWorkbookBuilder() {
-        workbook = new XSSFWorkbook();
+  private final XSSFWorkbook workbook;
+  private XSSFSheet dataSheet;
+  private XSSFSheet pivotTableSheet;
+  private XSSFPivotTable pivotTable;
+  private AreaReference source;
+  private final List<Field<T>> fields = new ArrayList<>();
+  private Collection<T> data;
+
+  private ReportWorkbookBuilder() {
+    workbook = new XSSFWorkbook();
+  }
+
+  public static <T> ReportWorkbookBuilder<T> newWorkbook(Class<T> klass) {
+    return new ReportWorkbookBuilder<>();
+  }
+
+  public ReportWorkbookBuilder<T> fieldText(String header, Function<T, ? extends Object> mapper) {
+    return field(header, CellType.STRING, mapper);
+  }
+
+  public ReportWorkbookBuilder<T> fieldNum(String header, Function<T, ? extends Object> mapper) {
+    return field(header, CellType.NUMERIC, mapper);
+  }
+
+  private ReportWorkbookBuilder<T> field(String header, CellType cellType, Function<T, ? extends Object> mapper) {
+    fields.add(Field.newField(header, cellType, mapper));
+    return this;
+  }
+
+  public ReportWorkbookBuilder<T> data(Collection<T> data) {
+    this.data = data;
+    dataSheet = dataSheet == null
+        ? workbook.createSheet("CDL_Data")
+        : dataSheet;
+    int rowCount = 0;
+    addDataHeaderRow(dataSheet.createRow(rowCount++), fields);
+    for (T datum : data) {
+      addDataRow(dataSheet.createRow(rowCount++), datum);
     }
+    return this;
+  }
 
-    public static <T> ReportWorkbookBuilder<T> newWorkbook(Class<T> klass) {
-        return new ReportWorkbookBuilder<>();
-    }
-
-    public ReportWorkbookBuilder<T> fieldText(String header, Function<T, ? extends Object> mapper) {
-        return field(header, CellType.STRING, mapper);
-    }
+  private static final char[] ALPHABET = {
+    'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M',
+    'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U', 'V', 'W', 'X', 'Y', 'Z'};
 
-    public ReportWorkbookBuilder<T> fieldNum(String header, Function<T, ? extends Object> mapper) {
-        return field(header, CellType.NUMERIC, mapper);
-    }
+  public String getLastDataColumnLetter() {
+    return getLastDataColumnLetter(fields);
+  }
 
-    private ReportWorkbookBuilder<T> field(String header, CellType cellType, Function<T, ? extends Object> mapper) {
-        fields.add(Field.newField(header, cellType, mapper));
-        return this;
+  public static String getLastDataColumnLetter(List fields) {
```

Comment:
You need comments here - and for all public classes. 

I believe this does not work. If I understand what it is doing, it should require two arithmetic expressions - not a loop and no pow() function call.